dotCMS/src/main/java/com/dotcms/saml/DotSamlProxyFactory.java

```diff
@@ -0,0 +1,133 @@
+package com.dotcms.saml;
+
+import com.dotcms.saml.service.external.IdentityProviderConfigurationFactory;
+import com.dotcms.saml.service.external.MessageObserver;
+import com.dotcms.saml.service.external.SamlAuthenticationService;
+import com.dotcms.saml.service.external.SamlConfigurationService;
+import com.dotcms.saml.service.external.SamlException;
+import com.dotcms.saml.service.external.SamlServiceBuilder;
+import com.dotcms.saml.service.impl.SamlServiceBuilderImpl;
+import com.dotcms.security.apps.AppDescriptor;
+import com.dotcms.security.apps.AppsAPI;
+import com.dotmarketing.business.APILocator;
+import com.liferay.portal.model.User;
+import io.vavr.control.Try;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * This is the proxy to provides the object to interact with the Saml Osgi Bundle
+ *
+ * @author jsanca
+ */
+public class DotSamlProxyFactory {
+
+    public static final String SAML_APP_CONFIG_KEY = "dotsaml-config";
+
+    private final SamlServiceBuilder samlServiceBuilder = new SamlServiceBuilderImpl(); // todo: get this from osgi reference
+    private final MessageObserver    messageObserver    = new DotLoggerMessageObserver();
+    private final AppsAPI            appsAPI            = APILocator.getAppsAPI();
+    private final SamlConfigurationService samlConfigurationService =
+            new SamlConfigurationServiceImpl();
+    private final IdentityProviderConfigurationFactory identityProviderConfigurationFactory =
```

Comment:
Codacy found an issue: [Field identityProviderConfigurationFactory has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)